app/src/main/java/org/mozilla/fenix/settings/wallpaper/WallpaperSettings.kt

```diff
@@ -0,0 +1,163 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.settings.wallpaper
+
+import androidx.compose.foundation.BorderStroke
+import androidx.compose.foundation.ExperimentalFoundationApi
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.background
+import androidx.compose.foundation.border
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.aspectRatio
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.lazy.GridCells
+import androidx.compose.foundation.lazy.LazyVerticalGrid
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.Surface
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import mozilla.components.lib.state.ext.observeAsComposableState
+import org.mozilla.fenix.R
+import org.mozilla.fenix.components.AppStore
+import org.mozilla.fenix.components.appstate.AppAction
+import org.mozilla.fenix.theme.FirefoxTheme
+import org.mozilla.fenix.wallpapers.Wallpaper
+
+/**
+ * A screen for various settings related to wallpapers.
+ *
+ * @param wallpaperManager Manager for retrieving and updating current wallpaper.
+ */
+@Composable
+fun WallpaperSettings(
+    appStore: AppStore
```

Comment:
We do not want to pass in the AppStore into any @Composable function. The reason for this is that it will react to any changes in the AppStore compare to just the wallpaper state. You will want to `observeAsComposableState` prior to calling the Composable content https://github.com/mozilla-mobile/fenix/pull/23113/files#diff-ecfc0b5554895ea8d6552fbd4096ae0b65cb027f3d6bd6e854f9a673ef59c9a8R30. See https://searchfox.org/mozilla-mobile/source/fenix/app/src/main/java/org/mozilla/fenix/home/recentbookmarks/view/RecentBookmarksViewHolder.kt#32

Reply:
I'm confused as to the difference, could you explain it in a little more detail? My understanding was that in either case, we're performing (compose) `State` reads in a `Composition`, so we would get smart recompositions down the UI tree. Why is observing the `Store` in one level above preferred?

Reply:
I think this was the best resource I found to describe my intentions. https://developer.android.com/jetpack/compose/state#state-hoisting

The composable caller in this cause happens in https://github.com/mozilla-mobile/fenix/pull/23113/files#diff-ecfc0b5554895ea8d6552fbd4096ae0b65cb027f3d6bd6e854f9a673ef59c9a8R30. 

Otherwise, I am seeing that in `WallpaperSettings`, you are already rendering UI, you don't necessarily want to pass and duplicate the AppStore, you may want to make this composable more generic and not necessarily depend on the AppStore because all it cares about is the `wallpaper` state.